### PR TITLE
Allow null for PaletteManipulator parent

### PR DESCRIPTION
**Pull request by fritzmg**
#5032 for Contao 5.0.

### CODE REVIEW

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/DataContainer/PaletteManipulator.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public static function create(): self
      *
      * @throws PalettePositionException
      */
-    public function addLegend(string $name, array|string $parent, string $position = self::POSITION_AFTER, bool $hide = false): self
+    public function addLegend(string $name, array|string $parent = null, string $position = self::POSITION_AFTER, bool $hide = false): self
```
Comment (leofeyer) [inline core-bundle/src/DataContainer/PaletteManipulator.php]:
```suggestion
    public function addLegend(string $name, array|string|null $parent = null, string $position = self::POSITION_AFTER, bool $hide = false): self
```

File: core-bundle/src/DataContainer/PaletteManipulator.php [hunk 2]
```diff
@@ -54,7 +54,7 @@ public function addLegend(string $name, array|string $parent, string $position =
      *
      * @throws PalettePositionException
      */
-    public function addField(array|string $name, array|string $parent, string $position = self::POSITION_AFTER, \Closure|array|string|null $fallback = null, string $fallbackPosition = self::POSITION_APPEND): self
+    public function addField(array|string $name, array|string $parent = null, string $position = self::POSITION_AFTER, \Closure|array|string|null $fallback = null, string $fallbackPosition = self::POSITION_APPEND): self
```
Comment (leofeyer) [inline core-bundle/src/DataContainer/PaletteManipulator.php]:
```suggestion
    public function addField(array|string $name, array|string|null $parent = null, string $position = self::POSITION_AFTER, \Closure|array|string|null $fallback = null, string $fallbackPosition = self::POSITION_APPEND): self
```

**Review by leofeyer — CHANGES_REQUESTED** [review]

**Review by ausi — APPROVED** [review]

**Comment by leofeyer** [comment]
What am I supposed to review? 😄 I guess the changes have already been merged upstream on Monday.

**Comment by fritzmg** [comment]
Ah yes, did not realise. Also did not know these changes were able to be merged upstream :) (that's why I did the separate PR for 5.0 in the first place). Closing then.

**Pull request closed without merging**